### PR TITLE
Feat/multi batch processing

### PR DESCRIPTION
**Pull request by bartoszptak**

### CODE REVIEW

File: src/deepness/common/processing_parameters/map_processing_parameters.py [hunk 1]
```diff
@@ -37,6 +37,8 @@ class MapProcessingParameters:
     resolution_cm_per_px: float  # image resolution to used during processing
     processed_area_type: ProcessedAreaType  # whether to perform operation on the entire field or part
     tile_size_px: int  # Tile size for processing (model input size)
+    batch_size: int  # Batch size for processing
+    local_cache: bool  # Whether to use local cache for tiles
```
Comment (przemyslaw-aszkowski) [inline src/deepness/common/processing_parameters/map_processing_parameters.py:41]: How about adding in comment it is caching on disk?

**Review by przemyslaw-aszkowski — APPROVED** [review]

File: test/test_map_processor_detection_oils_example.py [hunk 2]
```diff
@@ -52,6 +54,38 @@ def test_map_processor_detection_oil_example():
...
+    )
+
+    map_processor = MapProcessorDetection(
+        rlayer=rlayer,
+        vlayer_mask=None,
+        map_canvas=MagicMock(),
+        params=params,
+    )
+
+    map_processor.run()
```
Comment (przemyslaw-aszkowski) [inline test/test_map_processor_detection_oils_example.py:88]: How about adding an assert for the result (also in the original non-batched test)

File: test/test_map_processor_segmentation_landcover_example_batched.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        rlayer=rlayer,
+        vlayer_mask=None,
+        map_canvas=MagicMock(),
+        params=params,
+    )
+
+    map_processor.run()
+    result_img = map_processor.get_result_img()
+
+    assert result_img.shape == (2351, 2068)
```
Comment (przemyslaw-aszkowski) [inline test/test_map_processor_segmentation_landcover_example_batched.py:54]: how about asserting some pixels also, to check the result?

**Review by przemyslaw-aszkowski — APPROVED** [review]

File: test/test_map_processor_segmentation_landcover_example_batched.py (hunk 3 above)
Comment (bartoszptak) [reply test/test_map_processor_segmentation_landcover_example_batched.py:56]: Check the last commit, please.

**Comment by bartoszptak** [comment]
Tests show poor acceleration with multi-batch processing. For example, large area map processing using RTX4090 results 5m11s (batch=1) vs 4m42s (batch=32).

Profiling results: 
image (1)

**Pull request merged**